### PR TITLE
feat: implements define to replace globals

### PR DESCRIPTION
**Pull request by cckrause**
This feature allows to define globals to be replaced with build time values:

```javascript
            htmlPlugin({
                files: [
                    {
                        entryPoints: [
                            'src/App.jsx',
                        ],
                        filename: `index.html`,
                        htmlTemplate: await fs.readFile('./src/index.html', 'utf-8'),
                        define: {
                            $$SOME_URL: 'https://test.foo.com',
                            __SOME_HTML_INJECTION__: '<div class="foo">hello world!</div>',
                        },
                    },
                ],
            }),
``` 

input html:
```
<!DOCTYPE html>
<html lang="de">
    <head>
    </head>

    <body>
       <a href="$$SOME_URL">some link</a>
        __SOME_HTML_INJECTION__
    </body>
</html>

```
results in:
```
<!DOCTYPE html>
<html lang="de">
    <head>
    </head>

    <body>
        <a href="https://test.foo.com">some link</a>
        <div class="foo">hello world!</div>
    </body>
</html>

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,20 +104,23 @@ esbuild.build(options).catch(() => process.exit(1))
...
 
 interface HtmlFileConfiguration {
-    filename: string,           // Output filename, e.g. index.html. This path is relative to the out dir
-    entryPoints: string[],      // Entry points to inject into the created html file, e.g. ['src/index.jsx']. 
-                                // Multiple entryPoints are possible.
-    title?: string,             // title to inject into the head, will not be set if not specified
-    htmlTemplate?: string,      // custom html document template string. If you omit a template, 
-                                // a default template will be used (see below)
+    filename: string,                     // Output filename, e.g. index.html. This path is relative to the out dir
+    define: HtmlFileConfigurationDefine,  // provide a way to replace global identifiers with constant expressions
```
Comment (cckrause) [inline README.md:115]:
```suggestion
    define: HtmlFileConfigurationDefine,  // To replace global identifiers with constant expressions
```

File: lib/cjs/index.d.ts [hunk 2]
```diff
@@ -6,8 +6,10 @@ interface HtmlFileConfiguration {
     filename: string;
     entryPoints: string[];
     title?: string;
-    htmlTemplate?: string;
+    htmlTemplate?: any;
```
Comment (cckrause) [inline lib/cjs/index.d.ts:9]:
```suggestion
    htmlTemplate?: string;
```

File: src/index.ts [hunk 3]
```diff
@@ -11,10 +11,13 @@ interface HtmlFileConfiguration {
     filename: string,
     entryPoints: string[],
     title?: string,
-    htmlTemplate?: string,
+    htmlTemplate?: any,
```
Comment (cckrause) [inline src/index.ts:14]:
```suggestion
    htmlTemplate?: string,
```

File: src/index.ts [hunk 4]
```diff
@@ -56,7 +65,7 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
                 }
                 logInfo && console.log()
 
-                const outdir = build.initialOptions.outdir!;
+                const outdir = build.initialOptions.outdir!;
```
Comment (cckrause) [inline src/index.ts:68]:
```suggestion
                const outdir = build.initialOptions.outdir!;
```

Comment (cckrause) [reply src/index.ts:68]: code cleanup

**Review by creadoit — APPROVED** [review]

**Review by dominiq-eu — APPROVED** [review]

**Comment by craftamap** [comment]
Hey :wave:,

As discussed, this will be superset with a templating solution using lodash.template (See #14). Thanks for your contribution regardless!

**Pull request closed without merging**